planning/behavior_path_planner_common/src/utils/drivable_area_expansion/drivable_area_expansion.cpp

```diff
@@ -121,52 +121,79 @@ double calculate_minimum_lane_width(
-        const auto dist = tier4_autoware_utils::calcDistance2d(*intersection_point, offset_point);
-        minimum_expansions[bound_idx] = std::max(minimum_expansions[bound_idx], dist);
-        minimum_expansions[bound_idx + 1] = std::max(minimum_expansions[bound_idx + 1], dist);
-        // apply the expansion to all bound points within the extra arc length
-        auto arc_length =
-          tier4_autoware_utils::calcDistance2d(*intersection_point, bound[bound_idx + 1]);
-        for (auto up_bound_idx = bound_idx + 2; up_bound_idx < bound.size(); ++up_bound_idx) {
-          arc_length +=
-            tier4_autoware_utils::calcDistance2d(bound[up_bound_idx - 1], bound[up_bound_idx]);
-          if (arc_length > params.extra_arc_length) break;
-          minimum_expansions[up_bound_idx] = std::max(minimum_expansions[up_bound_idx], dist);
-        }
-        arc_length = tier4_autoware_utils::calcDistance2d(*intersection_point, bound[bound_idx]);
-        for (auto down_offset_idx = 1LU; down_offset_idx < bound_idx; ++down_offset_idx) {
-          const auto idx = bound_idx - down_offset_idx;
-          arc_length += tier4_autoware_utils::calcDistance2d(bound[idx - 1], bound[idx]);
-          if (arc_length > params.extra_arc_length) break;
-          minimum_expansions[idx] = std::max(minimum_expansions[idx], dist);
-        }
-        break;
+      const auto prev_p = convert_point(bound[bound_idx]);
+      const auto next_p = convert_point(bound[bound_idx + 1]);
+      const auto projection = point_to_segment_projection(path_p, prev_p, next_p);
+      if (projection.distance < bound_projections[path_idx].distance) {
+        bound_indexes[path_idx] = bound_idx;
+        bound_projections[path_idx] = projection;
       }
     }
+    lb_idx = bound_indexes[path_idx];
+  }
+}
+
+void apply_extra_arc_length(
+  Expansion & expansion, const std::vector<Point> & bound, const double extra_arc_length,
+  const Side side)
+{
+  const auto & bound_indexes =
+    side == LEFT ? expansion.left_bound_indexes : expansion.right_bound_indexes;
+  const auto & bound_projections =
+    side == LEFT ? expansion.left_projections : expansion.right_projections;
+  auto & bound_expansions = side == LEFT ? expansion.left_distances : expansion.right_distances;
+  const auto original_expansions = bound_expansions;
+  for (auto path_idx = 0UL; path_idx < bound_indexes.size(); ++path_idx) {
+    const auto bound_idx = bound_indexes[path_idx];
+    auto arc_length = boost::geometry::distance(
+      bound_projections[path_idx].point, convert_point(bound[bound_idx + 1]));
+    for (auto up_bound_idx = bound_idx + 2; up_bound_idx < bound.size(); ++up_bound_idx) {
+      arc_length +=
+        tier4_autoware_utils::calcDistance2d(bound[up_bound_idx - 1], bound[up_bound_idx]);
+      if (arc_length > extra_arc_length) break;
+      bound_expansions[up_bound_idx] =
+        std::max(bound_expansions[up_bound_idx], original_expansions[bound_idx]);
+    }
```

Comment:
See if this can be put in a lambda to be reused for the down direction

Reply:
[76d713d](https://github.com/autowarefoundation/autoware.universe/pull/5640/commits/76d713d7e6603acfb6584b87e218171386ac6ed7)